core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/recoverer.go

```diff
@@ -596,3 +604,18 @@ func (r *logRecoverer) removePending(workID string) {
 	}
 	r.pending = updated
 }
+
+// sortPending sorts the pending list by a random order based on the latest block.
```

Comment:
this will be different order on nodes as latestBlock will be out of sync. We don't really need consensus in recoverables so it should be ok here, but in provider we can use `latestBlock % 100` maybe to ensure it stays same across nodes for some time

Reply:
Used `latestBlock / 10`, do you think we need 100? (maybe for fast chains)